sql_lineage/lineage.py

```diff
@@ -16,8 +18,12 @@ def _parse_ctes(sql: str) -> [Nodes, Edges]:
     """
     Parse an SQL query into nodes and edges based on CTEs.
     """
-    parsed = qualify.qualify(sqlglot.parse_one(sql))
     nodes, edges = [], []
+    parsed = sqlglot.parse_one(sql)
+    with contextlib.suppress(Exception):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Ensure there's a test covering the scenario where an exception is suppressed.

This change introduces a broad exception handling which could potentially suppress significant errors. It's crucial to have a test case ensuring that only the intended exceptions are caught and that the system behaves as expected in such cases.

```suggestion
    try:
        parsed = qualify.qualify(parsed)
    except Exception as e:
        logging.warning(f"Failed to qualify parsed SQL due to: {e}. Proceeding without qualification.")

    for cte in parsed.find_all(exp.CTE):
```

Reply:
Is this comment correct?

Reply:
Is this comment helpful?

Reply:
Is the comment type correct?

Reply:
Is the comment area correct?